kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -241,15 +242,13 @@ public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
 
     @Override
     public void onDeploy(DeploymentEvent event) {
-        event.getDeployedUnit().getRuntimeManager().getRuntimeEngine(ProcessInstanceIdContext.get()).getKieSession()
-                .addEventListener(this);
+        ((InternalRegisterableItemsFactory) ((InternalRuntimeManager) event.getDeployedUnit().getRuntimeManager())
+                .getEnvironment().getRegisterableItemsFactory()).addProcessListener(this);
         updateRegistration(event, this::updateTopics);
     }
 
     @Override
     public void onUnDeploy(DeploymentEvent event) {
-        event.getDeployedUnit().getRuntimeManager().getRuntimeEngine(ProcessInstanceIdContext.get()).getKieSession()
-                .removeEventListener(this);
```

Comment:
When Kjar is undeployed, the list of listeners is clear up. So when ondeploy is invoked againd, the list will be empty
Actually if we try to remove the listener on undeploy, we will get an exception because the deployment is already closed